haskell-src/Concordium/Types/Transactions.hs

```diff
@@ -190,6 +191,141 @@ instance HashableTo TransactionHashV0 AccountTransaction where
 instance HashableTo TransactionSignHashV0 AccountTransaction where
     getHash = atrSignHash
 
+-- | The signatures for an 'AccountTransactionV1'.
+--
+--  * @SPEC: <$DOCS/Transactions#transaction-signatures-v1>
+data TransactionSignaturesV1 = TransactionSignaturesV1
+    { -- | The signatures for the sender account
+      tsv1Sender :: TransactionSignature,
+      -- | The signatures for the sponsor account. These must be present if a sponsor
+      -- is specified for the transaction in the corresponding transaction header.
+      tsv1Sponsor :: Maybe TransactionSignature
+    }
+    deriving (Show, Eq)
+
+instance S.Serialize TransactionSignaturesV1 where
+    put TransactionSignaturesV1{..} =
+        S.put tsv1Sender <> S.put tsv1Sponsor
+
+    get = S.label "transaction signatures v1" $ do
+        tsv1Sender <- S.label "sender" S.get
+        tsv1Sponsor <- S.label "sponsor" S.get
+        return $! TransactionSignaturesV1{..}
+
+-- | Data common to all v1 transaction types.
+--
+--  * @SPEC: <$DOCS/TransactionsV1#transaction-header-v1>
+data TransactionHeaderV1 = TransactionHeaderV1
+    { -- | Sender account.
+      thv1Sender :: AccountAddress,
+      -- | Account nonce.
+      thv1Nonce :: !Nonce,
+      -- | Amount of energy dedicated for the execution of this transaction.
+      thv1EnergyAmount :: !Energy,
+      -- | Size of the payload in bytes.
+      thv1PayloadSize :: PayloadSize,
+      -- | Absolute expiration time after which transaction will not be executed
+      thv1Expiry :: TransactionExpiryTime,
+      -- | An optional sponsor account which pays the transaction fees for the
+      -- transtaction execution
+      thv1Sponsor :: Maybe AccountAddress
+    }
+    deriving (Show, Eq)
+
+transactionHeaderV0 :: TransactionHeaderV1 -> TransactionHeader
```

Comment:
Suggest to name this a bit more descriptive, something like `toTransactionHeader`.